### PR TITLE
Treatments map

### PR DESCRIPTION
**Pull request by pflopez**
This pr adds the maplibre map on treatments and project areas, showing stands.
Additionally, adds some stubs/temporary components to interact with the map

TODO, not included on this pr:
- Loading the project areas geojson on the map (similar to scenario results)
- Assign prescriptions
- All UI work for this screens

Note, might need to do `npm install --legacy-peer-deps` to avoid errors locally. Hopefully using this flag would be temporary as there is already a change to fix this on its way https://github.com/Esri/esri-leaflet-vector/pull/219#issuecomment-2293790731

Additionally the map libre libraries are not included on the main bundle

Screen Shot 2024-08-16 at 11 23 49
Screen Shot 2024-08-16 at 11 24 14

### CODE REVIEW

File: src/interface/src/app/treatments/prescription-actions/prescription-actions.component.html [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<div *ngIf="prescriptions$ | async; let prescriptions" class="prescriptions">
+  <button *ngFor="let action of prescriptions.SINGLE | keyvalue" [value]="">
```
Comment (lastminutediorama) [inline src/interface/src/app/treatments/prescription-actions/prescription-actions.component.html:2]: did you mean to leave value blank here?

File: src/interface/src/app/treatments/map-rectangle/map-rectangle.component.html [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<mgl-geojson-source id="rectangle-source">
+  <mgl-feature [geometry]="rectangleGeometry"></mgl-feature>
+</mgl-geojson-source>
+<mgl-layer
+  id="selection-rectangle"
+  type="line"
+  [source]="sourceName"
+  [paint]="{
+    'line-color': '#007dff',
```
Comment (lastminutediorama) [inline src/interface/src/app/treatments/map-rectangle/map-rectangle.component.html:9]: I don't know if it's worth the effort yet, but if there's a way to store these styles as constants somewhere, maybe that would help us as this grows?

File: src/interface/src/app/treatments/treatment-map/treatment-map.component.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  ],
+  providers: [SelectedStandsState],
+  templateUrl: './treatment-map.component.html',
+  styleUrl: './treatment-map.component.scss',
+})
+export class TreatmentMapComponent {
+  mapLibreMap!: MapLibreMap;
+  readonly key = environment.stadiamaps_key;
+
+  // ew, not sure if I want to do prop drilling here or set up something else (provider?)
```
Comment (lastminutediorama) [inline src/interface/src/app/treatments/treatment-map/treatment-map.component.ts:43]: I think this is okay for now, but maybe we can talk about what a provider could look like, especially if we're talking about where and how we're storing these and presenting them to the frontend?

**Review by lastminutediorama — APPROVED** [review]
Just some general questions, but nothing that needs to be addressed. And since you mentioned it, I didn't have any better ideas for naming -- I think the components, functions, and vars are all self-explanatory and don't risk being confused with anything else.

File: src/interface/src/app/treatments/treatment-map/treatment-map.component.ts (hunk 3 above)
Comment (pflopez) [reply src/interface/src/app/treatments/treatment-map/treatment-map.component.ts:43]: sure! I also think its ok now, and i don't want to prematurely optimize until we have this a little bit more developed.
A provider here can be something similar to what we have for the stands (SelectedStandsState).

Will update the comment though that one was mostly for me :D

File: src/interface/src/app/treatments/map-rectangle/map-rectangle.component.html (hunk 2 above)
Comment (pflopez) [reply src/interface/src/app/treatments/map-rectangle/map-rectangle.component.html:9]: Yes i think that's probably the next step once we have more layers

File: src/interface/src/app/treatments/prescription-actions/prescription-actions.component.html (hunk 1 above)
Comment (pflopez) [reply src/interface/src/app/treatments/prescription-actions/prescription-actions.component.html:2]: yeah that's a placeholder, im not doing assignments on this PR, to limit the scope of changes.

**Pull request merged**